agents/lakeshore372/LS372_agent.py

```diff
@@ -129,16 +153,24 @@ def start_acq(self, session, params=None):
                     time.sleep(.1)
 
                 else:
-                    active_channel = self.module.get_active_channel()
-
-                    # The 372 reports the last updated measurement repeatedly
-                    # during the "pause change time", this results in several
-                    # stale datapoints being recorded. To get around this we
-                    # query the pause time and skip data collection during it
-                    # if the channel has changed (as it would if autoscan is
-                    # enabled.)
-                    if previous_channel != active_channel:
-                        if previous_channel is not None:
+
+                    with self._lock_acquire('acq', 10.) as acquired:
```

Comment:
I'm not sure about what the behavior is for when multiple threads are waiting for the same lock. Right now there is no sleep time between when the cm drops the lock and attempts to re-acquire it. If another thread is waiting for that lock, are we sure it'll always be able to take it? Or is it random whether the lock will go to the waiting vs the acq process trying to re-acquire? Maybe we should add a sleep statement here just to be sure...